docs/configuration/tools/router_check.rst

```diff
@@ -0,0 +1,170 @@
+.. _config_tools_router_check_tool:
+
+Route table check tool
+======================
+
+**NOTE: The following configuration is for the route table check tool only and is not part of the Envoy binary.
+The route table check tool is a standalone binary that can be used to verify Envoy's routing for a given configuration
+file.**
+
+The following specifies input to the route table check tool. The route table check tool checks if
+the route returned by a :ref:`router <config_http_conn_man_route_table>` matches what is expected.
+The tool can be used to check cluster name, virtual cluster name,
+virtual host name, manual path rewrite, manual host rewrite, path redirect, and
+header field matches. Extensions for other test cases can be added. Details about installing the tool
+and sample tool input/output can be found at :ref:`installation <install_tools>`.
+
+The route table check tool config is composed of an array of json test objects. Each test object is composed of
+three parts.
+
+Test name
+  This field specifies the name of each test object.
+
+Input values
+  The input value fields specify the parameters to be passed to the router. Example input fields include
+  the :authority, :path, and :method header fields. The :authority and :path fields specify the url
+  sent to the router and are required. All other input fields are optional.
+
+Validate
+  The validate fields specify the expected values and test cases to check. At least one test
+  case is required.
+
+A simple tool configuration json has one test case and is writen as follows. The test
+expects a cluster name match of "instant-server".::
+
+   [
+     {
+       "test_name: "Cluster_name_test",
+       "input":
+         {
+           ":authority":"api.lyft.com",
+           ":path": "/api/locations"
+         }
+       "_validate"
```

Comment:
Now I realize that the comment recommended __ (two underscores). Is there a preference for two, one, or no prefix?